rasa_nlu/utils/spacy_utils.py

```diff
@@ -45,34 +45,35 @@ def required_packages(cls) -> List[Text]:
         return ["spacy"]
 
     @classmethod
-    def create(cls, cfg: RasaNLUModelConfig) -> 'SpacyNLP':
+    def create(cls,
+               component_config: Dict,
+               config: RasaNLUModelConfig) -> 'SpacyNLP':
         import spacy
 
-        component_conf = cfg.for_component(cls.name, cls.defaults)
-        spacy_model_name = component_conf.get("model")
+        spacy_model_name = component_config.get("model")
 
         # if no model is specified, we fall back to the language string
         if not spacy_model_name:
-            spacy_model_name = cfg.language
-            component_conf["model"] = cfg.language
+            spacy_model_name = config.language
+            component_config["model"] = config.language
 
         logger.info("Trying to load spacy model with "
                     "name '{}'".format(spacy_model_name))
 
         nlp = spacy.load(spacy_model_name, disable=['parser'])
         cls.ensure_proper_language_model(nlp)
-        return SpacyNLP(component_conf, nlp)
+        return cls(component_config, nlp)
 
     @classmethod
-    def cache_key(cls, model_metadata: 'Metadata') -> Text:
-
-        component_meta = model_metadata.for_component(cls.name)
+    def cache_key(cls,
+                  component_meta: Dict,
+                  model_metadata: 'Metadata') -> Optional[Text]:
 
         # Fallback, use the language name, e.g. "en",
         # as the model name if no explicit name is defined
         spacy_model_name = component_meta.get("model", model_metadata.language)
 
-        return cls.name + "-" + spacy_model_name
+        return cls.__name__ + "-" + spacy_model_name
```

Comment:
I think instead of `.__name__` we should use the `component.name()` function everywhere (e.g. here, but there is a couple more locations, please search for `.__name__` in pycharm)